Dynatrace/dynatrace-operator
# Reworked phase/status-field handling

# Description

In our current implementation the phase dynakube status field should indicate if all oneagents were deployed successfully. This was not the case because our code was searching for an old daemonset name which was no longer there. Therefore if the field was once set to error, it was never reseted. The bug was fixed in this PR.

Additionally, the operator and the deployment of the dynakube is more than just generating oneagents => Advanced the phase field checking: Now, the dynakube status field is 
- set to error if
  - One token is invalid
  - An reconciliation error occurs
- set to deploying if 
  - oneagent or activegate pods are not ready yet
- set to running if
  - none of the other cases happend

To quickly show the state of the deployment, the phase field was added to `additionalPrinterColumns` in our dynakube crd. 

Besides of that, a reconciliation of the dynakube was reworked a little but, to decrease the number of reconciliation the operator needs and unit tests for the dynakube reconciler were added.

## How can this be tested?

Apply the new CRD and start in a monitoring mode you prefer.
In the operator logs you can see: 
- Less reconciliations
- The state changes of the phase field

With `kubectl get dynakube` you should see the additional printer column.


## Checklist
